pkg/controller/configmap/configmap.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 Hewlett Packard Enterprise Development LP
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configmap
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	kdv1 "github.com/bluek8s/kubedirector/pkg/apis/kubedirector/v1beta1"
+	"github.com/bluek8s/kubedirector/pkg/observer"
+	"github.com/bluek8s/kubedirector/pkg/shared"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// ConfigMapType is a label placed on desired comfig maps that
+	// we want to watch and propogate inside containers
+	configMapType = shared.KdDomainBase + "/cmType"
+)
+
+var (
+	// StatusGens is exported so that the validator can have access
+	// to the ConfigMap StatusGens
+	StatusGens = shared.NewStatusGens()
+)
+
+// syncConfigMap runs the reconciliation logic. It is invoked because of a
+// change in or addition of configmap instance, currently there is no
+// polling for this resource. If the configmap is not labeled
+// with key ConfigMapType then no op
+func (r *ReconcileConfigMap) syncConfigMap(
+	reqLogger logr.Logger,
+	configmap *corev1.ConfigMap,
+) error {
+	// Memoize state of the incoming object.
+	oldMap, _ := observer.GetConfigMap(configmap.Namespace, configmap.Name)
+
+	// Set a defer func to write new status and/or finalizers if they change.
```

Comment:
This comment isn't right.

(Also there's no need to have this in a defer func at this point.)